docs/tutorial/understanding_ALF_via_the_minimal_working_example.rst

```diff
@@ -1,2 +1,425 @@
+is just the tip of the iceberg: a very sophisticated RL training pipeline runs
+under the surface. In this chapter, based on that minimal example, we will
+go through some major concepts of ALF and grasp the big picture.
+
+ALF design overview
+-------------------
+
+As a good analogy, you can think of ALF as a very flexible
+*circuit board*. A circuit board electrically connects electronic components using
+conductive tracks, and the various components on the board altogether fulfill some
+kind of complex function. Similarly, ALF is a framework that logically connects
+algorithmic components and fulfills RL training tasks.
+
+.. image:: images/alf_diagram.png
+    :width: 500
+    :align: center
+    :alt: ALF as a circuit board
+
+However, different with a circuit board, ALF's components are highly customizable. In the
+figure above, the temperature color (redness) represents how frequently each
+component is expected to be modified by a user in typical ALF use cases.
+
+.. note::
+    For examples:
+
+    1. If training an existing algorithm (e.g., SAC) on existing environments/tasks
+       (e.g., ``Hopper-v3``), only "data transformer" and "models&networks" are to
+       be modified.
+    2. If a new algorithm is tried on existing environments/tasks,
+       "algorithm hierarchy" is additionally to be modified.
+    3. In a rare case, "trainer" needs to be modified if the desired pipeline is
+       vastly different from what ALF has by default.
+
+Importantly, whatever component is changed, ALF's design makes sure that the
+pipeline still runs with the other components unchanged.
+
+And the customization is very easy to implement. The user only needs to write a
+configuration file in Python to specify which components need what changes. As a
+typical Python file, this conf file also supports writing new code for any ALF
+component (e.g., defining new algorithms, models, environments, etc).
+
+What the minimal example does
+-----------------------------
+
+Let's go back to our first example which trains an existing algorithm
+:class:`.ActorCriticAlgorithm` on ``CartPole-v0``. To achieve this, normally we would
+need to configure the environment name by
+
+.. code-block:: python
+
+    alf.config("create_environment",
+               env_name="CartPole-v0",
+               num_parallel_environments=30)
+
+which tells ALF to use ``CartPole-v0`` and create 30 environments in parallel for
+rollout data collection. The :func:`.create_environment` function is defined as:
+
+.. code-block:: python
+
+    @alf.configurable
+    def create_environment(env_name='CartPole-v0',
+                           env_load_fn=suite_gym.load,
+                           num_parallel_environments=30,
+                           nonparallel=False,
+                           seed=None,
+                           batched_wrappers=())
+
+We can see that because the default values of ``env_name`` and ``num_parallel_environments``
```

Comment:
Probably worth pointing out that for readability purpose, it is usually a good idea to not skip `env_name` and `num_parallel_environments`. `env_load_fn` on the other hand is usually skipped.